tests/profiling/conftest.py

```diff
@@ -0,0 +1,117 @@
+"""Some pytest configuration specific to the profiling test suite."""
+
+from importlib import resources
+
+import numpy as np
+import pytest
+import xarray
+
+
+@pytest.fixture(scope="session")
+def pmodel_profile_data(pytestconfig):
+    """Setting up the pmodel and loading the test data set.
+
+    This uses a module scoped yield fixture to only load the data once and then to
+    delete it all when the tests exit.
+    """
+    from pyrealm.pmodel import PModelEnvironment
+
+    print("\n***\nSetting up profiling of pmodel profiling data\n***\n")
+
+    # Loading the dataset:
+    dpath = resources.files("pyrealm_build_data") / "inputs_data_reduced_1y.nc"
+    ds = xarray.load_dataset(dpath)
+
+    # TODO - this is a bit of a hack because of the current unfriendly handling of data
+    #        that does not form neat blocks of daily data in the subdaily module. The
+    #        test data is a slice along longitude 24.25°E (Finland --> Crete -->
+    #        Botswana) so the actual local times needed for the subdaily module are
+    #        offset from the UTC times in the data. This step reduces the input data to
+    #        complete daily blocks of data using local time
+
+    ds = ds.sel(time=slice("2000-01-01T01:59", "2000-12-31T01:59"))
+    ds["local_time_offset"] = (ds["lon"] / 15 * 3600).astype("timedelta64[s]")
+    # 15° per hour, 3600 seconds per hour
+    ds["local_time"] = ds["time"] - ds["local_time_offset"]
+
+    # Function to increase the size of the test data by broadcasting identical
+    # values along the longitude axis - scaling along the years axis is more problematic
+    # because the years need to be properly calendar aware for subdaily models and so
+    # would need to handle leap year sequencing. Actually - no it isn't. You'd just
+    # calculate the number of days and then create the dates as a sequence from the
+    # first date - there would be some days that change year but it would be very very
+    # minor and not really relevant to this test.
+
+    def _scale_up(array, scaling_factor: int):
+        # Broadcast along the last axis
+        return np.broadcast_to(array, list(array.shape[:-1]) + [scaling_factor])
```

Comment:
Is there a reason why `_scale_up` here uses `broadcast_to` while `_scale_up` in `splash_profile_data` uses `tile`?

Reply:
I was scaling on longitude _and_ time - can broadcast one but then need tile to do both (although my broadcast-fu is not strong). I've dropped back to the simple case for now - it would be more efficient to broadcast, I think.

Reply:
Oh. Right. @tztsai The reason is that the `pmodel` data has shape (X, Y, 1) - that last singleton axis can be broadcast to (X, Y, N). But the `splash` data is (X, 20, 20) and that doesn't broadcast to (X, 20, 160) but does tile.

There may be a cleverer way to do this!